learn-ocaml.el

```diff
@@ -41,18 +44,29 @@
 	 (list (list learn-ocaml-command-name command token-option server-option id-option html-option dont-submit-option file nickname secret)))
     (cl-remove-if-not 'stringp list)))
 
+
+(defun learn-ocaml-yes-or-no (message)
+  (x-popup-dialog
+   t
+   `(,message
+     ("Yes" . t)
+     ("No" . nil))))
+  
 (defun learn-ocaml-error-handler (buffer callback proc string)
-  (if (not (string-equal string "finished\n"))
-      (message-box learn-ocaml-warning-message)
-    (when callback
-      (let ((result (if buffer
-			(progn
-			  (set-buffer buffer)
-			  (buffer-string))
-		      "")))
-	(when buffer (kill-buffer buffer))
-      (funcall callback result)))))
+  (let ((result (if buffer
+		    (progn
+		      (set-buffer buffer)
+		      (buffer-string))
+		  "")))
+    (when buffer (kill-buffer buffer))
+    (if (not (string-equal string "finished\n"))
```

Comment:
Here this is a typical pattern that can be simplified:
replacing `(if (not condition (progn i1 i2)))` with `(unless condition i1 i2)`

Reply:
I have a lot of 'if' that i can replace with then or unless in my code but here is not possible because there is an else

Reply:
indeed, sorry: the `if` cannot be avoided here (but maybe the `progn` can)